packages/@aws-cdk/aws-servicecatalog/test/tag-option.test.ts

```diff
@@ -0,0 +1,126 @@
+          key1: ['value1', 'value2'],
+          key2: ['value1', 'value2', 'value3'],
+        },
+      });
+
+      Template.fromStack(stack).resourceCountIs('AWS::ServiceCatalog::TagOption', 5);
+    }),
+
+    test('fails to create tag option with invalid minimum key length', () => {
+      expect(() => {
+        new servicecatalog.TagOptions(stack, 'TagOptions', {
+          tagOptions: {
+            '': ['value1', 'value2'],
+          },
+        });
+      }).toThrowError(/Invalid TagOption key for resource/);
+    }),
+
+    test('fails to create tag option with invalid maxium key length', () => {
+      expect(() => {
+        new servicecatalog.TagOptions(stack, 'TagOptions', {
+          tagOptions: {
+            ['longKey'.repeat(1000)]: ['value1', 'value2'],
+          },
+        });
+      }).toThrowError(/Invalid TagOption key for resource/);
+    }),
+
+    test('fails to create tag option with invalid value length', () => {
+      expect(() => {
+        new servicecatalog.TagOptions(stack, 'TagOptions', {
+          tagOptions: {
+            key: ['tagOptionValue'.repeat(1000)],
+          },
+        });
+      }).toThrowError(/Invalid TagOption value for resource/);
+    }),
+
+    test('associate tag options', () => {
+      const portfolio = new servicecatalog.Portfolio(stack, 'MyPortfolio', {
+        displayName: 'testPortfolio',
+        providerName: 'testProvider',
+      });
+
+      const tagOptions = new servicecatalog.TagOptions(stack, 'TagOptions', {
+        tagOptions: {
+          key1: ['value1', 'value2'],
+          key2: ['value1', 'value2', 'value3'],
+        },
+      });
+      portfolio.associateTagOptions(tagOptions);
+
+      Template.fromStack(stack).hasResource('AWS::ServiceCatalog::TagOption', 5);
+      Template.fromStack(stack).hasResource('AWS::ServiceCatalog::TagOptionAssociation', 5);
+    }),
+
+    test('create and associate tag options to different resources', () => {
+      const portfolio1 = new servicecatalog.Portfolio(stack, 'MyPortfolio1', {
+        displayName: 'testPortfolio1',
+        providerName: 'testProvider1',
+      });
+
+      const portfolio2 = new servicecatalog.Portfolio(stack, 'MyPortfolio2', {
+        displayName: 'testPortfolio2',
+        providerName: 'testProvider2',
+      });
+
+      const tagOptions = new servicecatalog.TagOptions(stack, 'TagOptions', {
+        tagOptions: {
+          key1: ['value1', 'value2'],
+          key2: ['value1', 'value2', 'value3'],
+        },
+      });
+
+      portfolio1.associateTagOptions(tagOptions);
+      portfolio2.associateTagOptions(tagOptions);
+
+      Template.fromStack(stack).hasResource('AWS::ServiceCatalog::TagOption', 5);
+      Template.fromStack(stack).hasResource('AWS::ServiceCatalog::TagOptionAssociation', 10);
+    }),
+
+    test('create and associate tag options in another stack', () => {
+      const stack1 = new cdk.Stack(app, 'Stack1');
+      const stack2 = new cdk.Stack(app, 'Stack2');
```

Comment:
Can we name these something more descriptive than `stack1` and `stack2`? `tagOptionsStack` and `productStack`, for example?